### PR TITLE
Generate IndexDocument from Publication

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -292,12 +292,7 @@
       <property name="allowedAnnotations" value="Override, Test"/>
       <property name="tokens" value="METHOD_DEF, CTOR_DEF, ANNOTATION_FIELD_DEF"/>
     </module>
-    <module name="MissingJavadocMethod">
```
Comment (axthosarouris) [inline config/checkstyle/checkstyle.xml:295]: stop complaining about methods with obvious implementation

File: search-commons/src/main/java/no/unit/nva/search/IndexPublisher.java [hunk 2]
```diff
@@ -2,12 +2,15 @@
...
 import com.fasterxml.jackson.annotation.JsonProperty;
+import no.unit.nva.model.Organization;
 import nva.commons.core.JacocoGenerated;
 
 import java.net.URI;
 import java.util.Objects;
 
 public class IndexPublisher {
+
+    private static final String NO_NAME = null;
```
Comment (axthosarouris) [inline search-commons/src/main/java/no/unit/nva/search/IndexPublisher.java:13]: I do not know where to get the name from

File: search-commons/src/main/java/no/unit/nva/search/PublicationToIndexDocumentMapper.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import no.unit.nva.model.EntityDescription;
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.Reference;
+import no.unit.nva.model.instancetypes.PublicationInstance;
+import no.unit.nva.search.IndexDocument.Builder;
+
+class PublicationToIndexDocumentMapper {
```
Comment (axthosarouris) [inline search-commons/src/main/java/no/unit/nva/search/PublicationToIndexDocumentMapper.java:15]: not public on purpose.

File: build.gradle [hunk 4]
```diff
@@ -49,7 +49,7 @@ subprojects {
     apply plugin: 'pmd'
 
     repositories {
-        jcenter()
+        mavenCentral()
```
Comment (axthosarouris) [inline build.gradle:52]: jcenter is obsolete

File: build.gradle [hunk 5]
```diff
@@ -59,6 +59,9 @@ subprojects {
     targetCompatibility = JavaVersion.VERSION_11 // bytecode target version
 
     dependencies {
+
+        testImplementation project(":test-commons")
```
Comment (axthosarouris) [inline build.gradle:63]: moved PublicationGenerator to a commons library.

**Review by brinxmat — APPROVED** [review]

**Pull request merged**